rust/blockstore/src/arrow/blockfile.rs

```diff
@@ -226,15 +226,13 @@ impl ArrowBlockfileWriter {
         if delta.get_size::<K, V>() > self.block_manager.max_block_size_bytes() {
             let new_blocks = delta.split::<K, V>(self.block_manager.max_block_size_bytes());
             for (split_key, new_delta) in new_blocks {
-                match self.root.sparse_index.add_block(split_key, new_delta.id) {
-                    Ok(_) => {
-                        let mut deltas = self.block_deltas.lock();
-                        deltas.insert(new_delta.id, new_delta);
-                    }
-                    Err(e) => {
-                        return Err(Box::new(e));
-                    }
-                };
+                self.root
+                    .sparse_index
+                    .add_block(split_key, new_delta.id)
+                    .map_err(|e| Box::new(e) as Box<dyn ChromaError>)?;
+
+                let mut deltas = self.block_deltas.lock();
+                deltas.insert(new_delta.id, new_delta);
```

Comment:
this change is equivalent to the old code, but figured I'd leave it in as it's a little cleaner